src/adhocracy_core/adhocracy_core/sheets/rate.py

```diff
@@ -189,20 +188,122 @@ class RateableSchema(colander.MappingSchema):
 )
 
 
-likeable_meta = rateable_meta._replace(
+class ILike(IPredicateSheet, ISheetReferenceAutoUpdateMarker):
+    """Marker interface for the like sheet."""
+
+
+class ILikeable(ISheet, ISheetReferenceAutoUpdateMarker):
+    """Marker interface for resources that can be liked."""
+
+
+class ICanLike(ISheet):
+    """Marker interface for resources that can like."""
+
+
+class CanLikeSchema(colander.MappingSchema):
+    """CanRate sheet data structure."""
+
+
+can_like_meta = sheet_meta._replace(isheet=ICanLike,
+                                    schema_class=CanLikeSchema)
+
+
+class LikeSubjectReference(SheetToSheet):
+    """Reference from like to liker."""
+
+    source_isheet = ILike
+    source_isheet_field = 'subject'
+    target_isheet = ICanLike
+
+
+class LikeObjectReference(SheetToSheet):
+    """Reference from like to liked resource."""
+
+    source_isheet = ILike
+    source_isheet_field = 'object'
+    target_isheet = ILikeable
+
+
+class LikeSchema(colander.MappingSchema):
+    """Like sheet data structure."""
+
+    subject = ReferenceSchema(reftype=LikeSubjectReference)
+    object = ReferenceSchema(reftype=LikeObjectReference)
+    like = Integer()
+
+    def validator(self, node, value):
+        """Validate the like."""
+        # TODO make this validator deferred
+        # TODO add post_pool validator
+        request = node.bindings['request']
+        _validate_subject_is_current_user(node, value, request)
+        _ensure_like_is_unique(node, value, request)
+        _query_registered_object_validator(node,
+                                           value,
+                                           request,
+                                           ILikeValidator,
+                                           'like')
+
+like_meta = sheet_meta._replace(isheet=ILike,
+                                schema_class=LikeSchema,
+                                sheet_class=AttributeResourceSheet,
+                                create_mandatory=True)
+
+
+@implementer(ILikeValidator)
+class LikeableLikeValidator:
+    """
+    Validator for rates about ILikeable.
+
+    The following values are allowed:
+
+      * 1: like
+      * 0: neutral/no vote
+    """
+
+    _allowed_values = (1, 0)
+
+    def __init__(self, context):
+        """Initialize self."""
+        self.context = context
+
+    def validate(self, rate: int) -> bool:
+        """Validate the rate."""
+        return rate in self._allowed_values
+
+    def helpful_error_message(self) -> str:
+        """Return error message."""
+        return 'rate must be one of {}'.format(self._allowed_values)
+
```

Comment:
With to sepated resource types/sheets we don't need the special validator adapter for rates anymore.
Its much easier to understand if we check allowed values in the colander schema.